Add docs on storage compaction concurrency (#18051)
cockroachdb/docs: src/current/_includes/v23.2/storage/compaction-concurrency.md

> By default, the [storage engine]({% link {{ page.version.version }}/architecture/storage-layer.md %}) uses a compaction concurrency of 4, with one slot dedicated to flushes. If you have sufficient IOPs and CPU headroom, you can consider increasing this setting via the `COCKROACH_COMPACTION_CONCURRENCY` environment variable for increased performance under some workloads. We **strongly recommend** testing your workload against non-default values of this setting.

Reviewer: The new `COCKROACH_COMPACTION_CONCURRENCY` environment variable doesn't include flushes (whereas the old undocumented rocksdb one did. So I think we should say "uses a compaction concurrency of 3", without referring to flushes.

Author: OK updated to 

> By default, the [storage engine]({% link {{ page.version.version }}/architecture/storage-layer.md %}) uses a compaction concurrency of 3.

> By default, the [storage engine]({% link {{ page.version.version }}/architecture/storage-layer.md %}) uses a compaction concurrency of 3. If you have sufficient IOPs and CPU headroom, you can consider increasing this setting via the `COCKROACH_COMPACTION_CONCURRENCY` environment variable. This may be useful in the following scenarios: it may help to reshape the LSM faster in [inverted LSM]({% link {{ page.version.version }}/architecture/storage-layer.md %}#inverted-lsms) scenarios; and it can lead to increased overall performance for some workloads. We **strongly recommend** testing your workload against non-default values of this setting.

Reviewer 2: ```suggestion
By default, the [storage engine]({% link {{ page.version.version }}/architecture/storage-layer.md %}) uses a compaction concurrency of 3. If you have sufficient IOPS and CPU headroom, you can consider increasing this setting via the `COCKROACH_COMPACTION_CONCURRENCY` environment variable. This may help to reshape the LSM more quickly in [inverted LSM]({% link {{ page.version.version }}/architecture/storage-layer.md %}#inverted-lsms) scenarios; and it can lead to increased overall performance for some workloads. Cockroach Labs **strongly recommends** testing your workload against non-default values of this setting.
```
